thrift/src/main/thrift/shared.thrift

```diff
@@ -182,4 +184,6 @@ struct Image {
   2: optional ImageAsset master
 
   3: required string mediaId
+
+  4: optional string source
```

Comment:
Composer includes that field when saving the trail image so we need it to pass this information to composer. The alternative would be query grid api before syncing the fields to composer but I think this solution is simpler.